acorn-to-esprima.js

```diff
@@ -7,9 +7,22 @@ exports.toToken = function (token) {
 
   if (type === tokTypes.name) {
     token.type = "Identifier";
-  } else if (type === tokTypes.semi || type === tokTypes.comma || type === tokTypes.parenL || type === tokTypes.parenR || type === tokTypes.braceL || type === tokTypes.braceR) {
+  } else if (type === tokTypes.semi ||
+             type === tokTypes.comma ||
+             type === tokTypes.parenL ||
+             type === tokTypes.parenR ||
+             type === tokTypes.braceL ||
+             type === tokTypes.braceR ||
+             type.isAssign) {
     token.type = "Punctuator";
-    token.value = type.type;
+    if (!token.value) {
```

Comment:
Because in the `type.isAssign` case, the `value` property already contains the operator, and `type.type` is undefined.